### ISSUE TITLE
Wrong probseg fmriprep output from freesurfer pipeline

### ISSUE THREAD

**Issue by julfou81** [nipreps/fmriprep#2387]
Hi, I noticed that the output of the segmentation from the freesurfer pipeline vs the FLIRT pipeline produced different probseg files. Especially, the probseg files from Freesurfer pipeline stream are binarized and not looking nice at all.
probseg-GM produced through Freesurfer pipeline:
![T1w_probseg_GM_FS](https://user-images.githubusercontent.com/32452352/111701344-fc75ee80-883a-11eb-97d6-a9a899e149e7.png)

probseg-GM produced through FSL  pipeline:
![T1w_probseg_GM_FSL](https://user-images.githubusercontent.com/32452352/111701444-262f1580-883b-11eb-9130-cceed15e8614.png)

probseg-GM + WM with Freesurfer pipeline:
![T1w_probseg_GM+WM_FS](https://user-images.githubusercontent.com/32452352/111701527-45c63e00-883b-11eb-877a-05dd4fbf5b9c.png)

probseg-GM + WM with FSL pipeline:
![T1w_probseg_GM+WM_FSL](https://user-images.githubusercontent.com/32452352/111701574-54acf080-883b-11eb-8fe2-b7b1ac184348.png)

probseg-WM with FS pipeline:
![T1w_probseg_WM_FS](https://user-images.githubusercontent.com/32452352/111702355-6fcc3000-883c-11eb-8263-04020d58155a.png)

wm.mgz file in the freesurfer folder: it looks ok!
![T1w_probseg_wmMGZ_freesurfer](https://user-images.githubusercontent.com/32452352/111702461-938f7600-883c-11eb-9f23-5fd5d0f3d4c5.png)

Has anybody also noticed this on its data?

 

### What version of fMRIPrep are you using?
20.2.0
### What kind of installation are you using? Containers (Singularity, Docker), or "bare-metal"?
Singularity
### What is the exact command-line you used?
```
</usr/local/miniconda/bin/fmriprep /work/EcritApp /work/EcritApp/derivatives --fs-license-file /work/freesurfer/license.txt participant --participant_label 001 --fd-spike-threshold 0.9 --return-all-components -w /work/temp_data_EcritApp --omp-nthreads 8 --nthreads 12 --mem_mb 40000 --ignore slicetiming --output-spaces T1w MNI152NLin2009cAsym MNIPediatricAsym:cohort-3>
```

### Have you checked that your inputs are BIDS valid?
Yes
### Can you find some traces of the error reported in the visual report (at the bottom) or in *crashfiles*?
```
<No errors to report!>
```

### Are you reusing previously computed results (e.g., FreeSurfer, Anatomical derivatives, work directory of previous run)?

Yes Freesurfer results were re-used

### fMRIPrep log
If you have access to the output logged by fMRIPrep, please make sure to attach it as a text file to this issue.

**Comment by julfou81** [comment]
This issue is still present for version 20.2.2. It looks that something is not working properly in  smriprep.workflows.anatomical, for the case with freesurfer: the part which is converting the .aseg file from freesurfer (which looks fine) to t1w_tpms is somehow doing something wrong.

**Comment by julfou81** [comment]
Here is another illustration. T1w image from 1 subject, preprocessed with FMRIPREP and normalized to MNI152NL2009cAsym.
1st case: with freesurfer processing, overlap of the tissue probability map of CSF:
![Capture d’écran 2021-07-20 à 10 09 17](https://user-images.githubusercontent.com/32452352/126285756-58ec9fb3-9285-44af-bd57-8aa9c87d40e5.png)
 
2nd case: same input T1w image, but this time, preprocessed with fmriprep with the flag ```--fs-no-reconall```:
![Capture d’écran 2021-07-20 à 10 10 28](https://user-images.githubusercontent.com/32452352/126286016-8ba3b431-478e-4d82-aec3-d4238464943c.png)

**Comment by oesteban** [comment]
Yes, I think we can agree on these tissue probability maps being poor. I think the one issue is that I never bothered to find the right way to generate tissue probability maps (as in GM, WM and CSF) from FreeSurfer outputs.

If someone more versed in FreeSurfer (@ahoopes?) could give me a hand, it would be very much appreciated. Right now we are just mapping labels from aseg to {WM, GM, CSF}.
